pkg/kubefed2/federate/federate.go

```diff
@@ -156,6 +165,14 @@ func FederateResource(hostConfig *rest.Config, qualifiedTypeName, qualifiedName
 		return nil, errors.Wrapf(err, "Error getting %s from %s %q", typeConfig.GetFederatedType().Kind, typeConfig.GetTarget().Kind, qualifiedName)
 	}
 
+	if outputYAML {
+		err := util.WriteUnstructuredToYaml(fedResource, cmdOut)
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to write federated resource to YAML")
+		}
+		return fedResource, nil
```

Comment:
Why does it make sense to return the resource if it has been output to yaml?  Note how `enable` separated generation of the resource from its creation in the `Run` method.  That also avoids the problem of cluttering the method signature of `FederateResource` with 2 variables with a very narrow use.

Reply:
The return param enables simplify (reading the same resource again in) the federate e2e test. Agree that the actual resource need not be returned for the yaml case here, updated to nil.
I am not sure if this breaks separation between functional and test code. If it does, let me know and I will update accordingly. 

To answer your suggestion about separating the generate and create, `FederateResource()` is actually doing that, however I did choose to use the bigger unit to write test on rather then using multiple smaller ones.